src/base/memory_monitor_sr.cpp

```diff
@@ -92,6 +92,77 @@ namespace cubmem
 
     return filecopy + ':' + std::to_string (line);
   }
+
+  void memory_monitor::add_stat (char *ptr, const size_t size, const char *file, const int line)
+  {
+    std::string tag_name;
+    MMON_METAINFO metainfo;
+
+    // size should not be 0 because of MMON_ALLOC_META_SIZE
+    assert (size > 0);
+
+    metainfo.allocated_size = (uint64_t) size;
+    m_total_mem_usage += metainfo.allocated_size;
+
+    tag_name = make_tag_name (file, line);
+
+    std::unique_lock <std::mutex> tag_map_lock (m_tag_map_mutex);
+    auto search = m_tag_map.find (tag_name);
+    if (search != m_tag_map.end ())
+      {
+	metainfo.tag_id = search->second;
+	m_stat_map[metainfo.tag_id] += metainfo.allocated_size;
+      }
+    else
+      {
+	metainfo.tag_id = m_tag_map.size ();
+	// tag_id is start with 0
+	std::pair <std::string, int> tag_map_entry (tag_name, metainfo.tag_id);
+	m_tag_map.insert (tag_map_entry);
+	m_stat_map.insert (std::make_pair (metainfo.tag_id, metainfo.allocated_size));
+      }
+    tag_map_lock.unlock ();
+
+    // put meta info into the allocated chunk
+    char *meta_ptr = ptr + metainfo.allocated_size - MMON_ALLOC_META_SIZE;
+    metainfo.magic_number = m_magic_number;
+    memcpy (meta_ptr, &metainfo, MMON_ALLOC_META_SIZE);
+    m_meta_alloc_count++;
+  }
+
+  void memory_monitor::sub_stat (char *ptr)
+  {
+    size_t allocated_size;
+
+    if (ptr == NULL)
+      {
+	return;
+      }
+
+#if defined(WINDOWS)
+    allocated_size = _msize ((void *)ptr);
+#else
+    allocated_size = malloc_usable_size ((void *)ptr);
+#endif // WINDOWS
```

Comment:
```suggestion
#if !defined(WINDOWS)
    allocated_size = malloc_usable_size ((void *)ptr);
#endif // !WINDOWS
```

how about defining the allocate_size only if it does not run on WINDOWS ?

Reply:
I change the code for Windows build with doing nothing.